TeamSnapSDK/SDK/CollectionJSON/TSDKCollectionObject.m

```diff
@@ -710,7 +715,10 @@ - (void)encodeWithCoder:(NSCoder *)coder {
 }
 
 - (BOOL)isNewObject {
-    return ([[self objectIdentifier] isEqualToString:@""] || [[self objectIdentifier] integerValue] == 0);
+    if (([[[self collection] data] objectForKey:@"id"] == nil) || [[[[[self collection] data] objectForKey:@"id"] description] isEqualToString:@""] || [[[[self collection] data] objectForKey:@"id"] isEqual:[NSNull null]] || ([[[[self collection] data] objectForKey:@"id"] integerValue] == 0)) {
+        return YES;
+    }
+    return FALSE;    
```

Comment:
return `NO` instead of `false` since this is objc

Reply:
yup. I like how I mixed YES and FALSE.